vhdbuilder/packer/install-dependencies.sh

```diff
@@ -459,6 +461,52 @@ elif [ "$OS" = "$MARINER_OS_NAME" ] && [ "$OS_VERSION" = "3.0" ] && [ "$(isARM64
   installAndConfigureArtifactStreaming acr-mirror-azurelinux3 rpm
 fi
 
+should_install_device_plugin() {
+  # Require x86_64 (skip arm64)
+  if [ "$(isARM64)" -eq 1 ]; then return 1; fi
+
+  # Ubuntu 24.04 only
+  if [ "$OS" = "$UBUNTU_OS_NAME" ] && [ "${UBUNTU_RELEASE:-}" = "24.04" ]; then return 0; fi
+
+  # Azure Linux 3.0 only
+  if [ "$OS" = "$AZURELINUX_OS_NAME" ] && [ "${OS_VERSION}" = "3.0" ]; then return 0; fi
+
+  return 1
+}
+
+install_device_plugin() {
+  local pkg="${K8S_DEVICE_PLUGIN_PKG}"
+  echo "Installing ${pkg} on ${OS} (Ubuntu ${UBUNTU_RELEASE:-n/a} / Version ${OS_VERSION:-n/a})..."
+  if [ "${OS}" = "${UBUNTU_OS_NAME}" ]; then
+    apt_get_install 30 1 600 "${pkg}" || exit $ERR_APT
+  elif [ "${OS}" = "${AZURELINUX_OS_NAME}" ]; then
+    if ! dnf_install 30 1 600 "${pkg}"; then
+      echo "Failed to install ${pkg} on Azure Linux ${OS_VERSION}" >&2
+      exit 1
+    fi
+  else
+    echo "install_device_plugin called for unsupported OS ${OS}" >&2
+    return 1
+  fi
+
+  echo "Disabling nvidia-device-plugin systemd unit..."
+  systemctl disable nvidia-device-plugin.service || exit 1
+  systemctl mask nvidia-device-plugin.service || exit 1
+
+  # Capture installed package version for logging
+  local pkg_version="unknown"
+  if [ "${OS}" = "${UBUNTU_OS_NAME}" ]; then
```

Comment:
Can we combine this if else into the previous one, so we have everything in one place? Or a switch case is also readable since we are only the same variable `$OS`

Reply:
Sure, combined it and used a switch case instead